src/checksum.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright 2013-2016, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use std::vec::Vec;
+use libc::size_t;
+use Checksum;
+use ffi;
+use translate::*;
+
+impl Checksum {
+    pub fn get_digest(self) -> Vec<u8> {
+        unsafe {
+            let mut digest_len: size_t = 4096;
+            let mut vec = Vec::with_capacity(digest_len as usize);
```

Comment:
Ah I see... You could store the type together with the GCheckSum I guess, but this is all becoming ugly. You could use the size of G_CHECKSUM_SHA512 for now... and we need to be careful when new variants are added.